<launch>
	<arg name="camera" 						default="4" />
	<arg name="name"						default="image$(arg camera)_pcl_fusion"/>
	<arg name="subscribe_topic_pcl" 		default="/velodyne/cloud" />
  	<arg name="subscribe_topic_image"		default="/ladybug5/camera$(arg camera)/image_color_rect" />
  	<arg name="subscribe_topic_image_info" 	default="/ladybug5/camera$(arg camera)/image_color_rect_info" />
  	<arg name="image_tf_frame_id"			default="ladybug_camera$(arg camera)" />
	<arg name="tf_buffer_length"			default="30" />
	<arg name="publish_pointcloud" 			default="$(arg subscribe_topic_image)/fused$(arg subscribe_topic_pcl)/cloud"/>
	<arg name="publish_image"	 			default="$(arg subscribe_topic_image)/fused$(arg subscribe_topic_pcl)/image"/>
	
	<arg name="image_frame_id"				default="ladybug_camera$(arg camera)"/>
	<arg name="reference_frame_id"			default="/odom"/>
	<arg name="nodelet_manager" 			default="nodelet_manager"/>
	<arg name="queue_size" 					default="30" />
	
	<!-- save images -->
	<arg name="save"						default="true" />
    <arg name="save_path" 					default="/home/fnolden/Bilder/exported/" />
	
	<node pkg="nodelet" type="nodelet" name="$(arg name)" args="load image_cloud/Fusion $(arg nodelet_manager)" output="screen">
	  	<param name="subscribe_topic_pcl" 			type="string" value="$(arg subscribe_topic_pcl)" />
	  	<param name="subscribe_topic_image" 		type="string" value="$(arg subscribe_topic_image)" />
	  	<param name="subscribe_topic_image_info" 	type="string" value="$(arg subscribe_topic_image_info)" />
	  	<param name="publish_topic_image"			type="string" value="$(arg publish_image)" />
	  	<param name="publish_topic_pcl"				type="string" value="$(arg publish_pointcloud)" />
	  	<param name="image_tf_frame_id"  			type="string" value="$(arg image_frame_id)" />
	  	<param name="reference_frame"  				type="string" value="$(arg reference_frame_id)" />
  		<param name="tf_buffer_length"							  value="$(arg tf_buffer_length)" />
  		<param name="queue_size" 								  value="$(arg queue_size)" />
	</node>
  
   	<include if="$(arg save)" file="$(find fiddle)/launch/extract_image.launch">
	  	<arg name="name" 	value="ex$(arg camera)_fusion" />
	  	<arg name="image" 	value="$(arg publish_image)" />	  	
	  	<arg name="path"	value="$(arg save_path)" />
	</include>
</launch>
